forem/forem
# Member index: Show applied roles filters

## What type of PR is this? (check all applicable)


## Description

We previously added functionality (behind a feature flag) to filter the member index by member roles. This PR updates the UI to show those applied filters on the index page itself (without having to open the filters modal), and allows users to remove each filter individually or "Clear all".

## Related Tickets & Documents



- Related Issue https://github.com/forem/forem/issues/17489
- Closes https://github.com/forem/forem/issues/17945

## QA Instructions, Screenshots, Recordings

- Enable the `member_index_view` feature flag and visit `/admin/member_manager/users`
- Open the filters modal and add some member roles filters, clicking "Apply filters" button
- Check your added filters appear on the index page
- Click a filter button to see it deleted and the page reloaded
- Click the "Clear all" link to check it clears all filters and reloads the page

NB: This UI section is hidden on the smallest viewport size (portrait mobile)


https://user-images.githubusercontent.com/20773163/174290010-f85ca498-9d39-4135-9930-0abdd4a4ef04.mp4


### UI accessibility concerns?

I went back and forth in my brain about whether these were buttons or links 🥴 

In the end I went with **buttons**, because...

Although _technically_ we direct the user to a new page on click (which means a link, right??) I felt like `button` was more consistent with user expectations. A link to "remove filter: admin" I felt is more confusing than a button with the same name. Plus, by this point the filtering mechanism of refreshing the page is known to the user, so hopefully the page reload would not be a surprise here.

Definitely happy to consider alternative perspectives on this though!

The buttons are also grouped together to help convey the related actions to screen reader users, and I've added a "visually hidden" heading to help screen reader users jump to this section more easily.


## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
- [X] This change does not need to be communicated, and this is why not: still behind a feature flag
